source/util/CommandLine.cpp

```diff
@@ -397,6 +413,34 @@ bool CommandLine::parse(span<const string_view> args, ParseOptions options) {
             continue;
         }
 
+        // check if arg is in the list of commands to skip
+        auto cmd_ignore_option = findOption("cmd_ignore");
```

Comment:
The CommandLine class should remain project-independent, meaning we could always extract it out and use it somewhere else without needing to change much, if anything. I think this gets a little too close to hardcoding slang-specific details inside this class. I think instead you could make this feature explicit in the interface of the class instead of implicit based on a magic name -- instead of a general add() for a new option, make a function like setIgnoreCommand() and setRenameCommand() and store those as separate members instead of the general list of options. Then you can avoid all this extra machinery and it will be a tiny bit more efficient to boot.

Reply:
OK, I understand, you suggest to make command renaming and ignoring an integral part of the CommandLine class.

Reply:
Yeah that's right, you basically already have done that here, so there's no reason not to make life easier on the implementation.

Reply:
Also just to say it, I think it's still an optional thing. We don't need to hardcode the name, the user of the class can enable it and tell you what name that command should have by doing something like enableIgnoreCommand("cmd_ignore"), enableReplaceCommand("cmd_replace")